Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -537,46 +538,78 @@ def validate_auth_header(headers: dict) -> bool:
     scehma: str = parts[0]
     jwt_token: str = parts[1]
     if scehma != 'Bearer' or not jwt_token:
+        demisto.info('Authorization header validation - failed to verify schema')
         return False
+
     decoded_payload: dict = jwt.decode(jwt_token, verify=False)
     issuer: str = decoded_payload.get('iss', '')
     if issuer != 'https://api.botframework.com':
+        demisto.info('Authorization header validation - failed to verify issuer')
+        return False
+
+    integration_context: dict = demisto.getIntegrationContext()
+    open_id_metadata: dict = json.loads(integration_context.get('open_id_metadata', '{}'))
+    keys: list = open_id_metadata.get('keys', [])
+
+    unverified_headers: dict = jwt.get_unverified_header(jwt_token)
+    key_id: str = unverified_headers.get('kid', '')
+    key_object: dict = dict()
+
+    # Check if we got the requested key in cache
+    for key in keys:
+        if key.get('kid') == key_id:
+            key_object = key
+            break
+
+    if not key_object:
+        # Didn't find requested key in cache, getting new keys
+        try:
+            open_id_url: str = 'https://login.botframework.com/v1/.well-known/openidconfiguration'
+            response: dict = requests.get(open_id_url).json()
+            jwks_uri: str = response.get('jwks_uri', '')
+            keys_response: dict = requests.get(jwks_uri).json()
+            keys = keys_response.get('keys', [])
+            open_id_metadata['keys'] = keys
+        except ValueError:
```

Comment:
Might want to catch also other exceptions. Pretty sure if there are network issues we will see other exceptions.